test/ecto/integration/crud_test.exs

```diff
@@ -107,23 +110,27 @@ defmodule Ecto.Integration.CrudTest do
         |> TestRepo.transaction()
     end
 
-    test "unsuccessful account creation" do
-      {:error, _, _, _} =
-        Ecto.Multi.new()
-        |> Ecto.Multi.insert(:account, fn _ ->
-          Account.changeset(%Account{}, %{name: nil})
-        end)
-        |> Ecto.Multi.insert(:user, fn _ ->
-          User.changeset(%User{}, %{name: "Bob"})
-        end)
-        |> Ecto.Multi.insert(:account_user, fn %{account: account, user: user} ->
-          AccountUser.changeset(%AccountUser{}, %{
-            account_id: account.id,
-            user_id: user.id
-          })
-        end)
-        |> TestRepo.transaction()
-    end
+    # "cannot open savepoint - SQL statements in progress"
```

Comment:
Are you sure it is prepared statements or that we aren't releasing a save point correctly?

Reply:
I added a log in `handle_transaction` to output the SQL and I couldn't see any case where we created a save point but did not release it. This error seems to be pretty rare, as I couldn't find anything via a search about it. If you look at `sqlite.c:88847` it mentions something about "active write statements", hence my hypothesis.

Reply:
Cool, I'll see if I can reliably recreate the issue and dig around a bit